src/VisualStudio/Core/Def/Implementation/ChangeSignature/AddParameterDialog.xaml.cs

```diff
@@ -0,0 +1,117 @@
+            ViewModel = new AddParameterDialogViewModel(notificationService);
+            _typeIntellisenseTextBoxView = typeIntellisenseTextBoxViewModel;
+            _nameIntellisenseTextBoxView = nameIntellisenseTextBoxViewModel;
+            _document = document;
+            this.Loaded += AddParameterDialog_Loaded;
+            DataContext = ViewModel;
+
+            InitializeComponent();
+        }
+
+        private void AddParameterDialog_Loaded(object sender, RoutedEventArgs e)
+        {
+            IntellisenseTextBox typeTextBox = new IntellisenseTextBox(
+                _typeIntellisenseTextBoxView, TypeContentControl);
+            this.TypeContentControl.Content = typeTextBox;
+
+            IntellisenseTextBox nameTextBox = new IntellisenseTextBox(
+                _nameIntellisenseTextBoxView, NameContentControl);
+            this.NameContentControl.Content = nameTextBox;
+        }
+
+        private void OK_Click(object sender, RoutedEventArgs e)
+        {
+            ViewModel.TypeName = ((IntellisenseTextBox)TypeContentControl.Content).Text;
+            ViewModel.ParameterName = ((IntellisenseTextBox)NameContentControl.Content).Text;
+
+            if (ViewModel.TrySubmit(_document))
+            {
+                DialogResult = true;
+            }
+        }
+
+        private void Cancel_Click(object sender, RoutedEventArgs e)
+        {
+            DialogResult = false;
+        }
+
+        private void TypeNameContentControl_PreviewKeyDown(object sender, KeyEventArgs e)
+        {
+            UIElement elementWithFocus = Keyboard.FocusedElement as UIElement;
+
+            if (elementWithFocus is IWpfTextView)
+            {
+                IntellisenseTextBox typeNameTextBox = elementWithFocus.GetParentOfType<IntellisenseTextBox>();
+
+                if (typeNameTextBox != null)
+                {
+                    if (e.Key == Key.Escape && !typeNameTextBox.HasActiveIntellisenseSession)
+                    {
+                        e.Handled = true;
+                    }
+                    else if (e.Key == Key.Enter && !typeNameTextBox.HasActiveIntellisenseSession)
+                    {
+                        // Do nothing. This case is handled in parent control KeyDown events.
+                    }
+                    else if (e.Key == Key.Tab && !typeNameTextBox.HasActiveIntellisenseSession)
+                    {
+                        // Do nothing. This case is handled in parent control KeyDown events.
+                    }
+                    else if (e.Key == Key.Space &&
+                        (typeNameTextBox.ContainerName.Equals("NameContentControl") ||
+                        (typeNameTextBox.ContainerName.Equals("TypeContentControl") && _document.Project.Language.Equals(LanguageNames.CSharp))))
+                    {
+                        // Do nothing. We disallow spaces in the name field for both C# and VB, and in the type field for C#.
```

Comment:
I can't type `Dictionary<string, string>` with a space in it?

Reply:
Agree. It was too strong. Remove. FYI: @dpoeschl